pydata_google_auth/auth.py

```diff
@@ -311,6 +312,19 @@ def get_user_credentials(
     pydata_google_auth.exceptions.PyDataCredentialsError
         If unable to get valid user credentials.
     """
+
+    # If the user isn't explicitly asking to authenticate with a particular
+    # client ID, then we should try to authenticate the user with Colab-based
+    # credentials, if possible.
+    if client_id is None:
+        # default_project ignored for colab credentials
+        credentials, _ = get_colab_default_credentials(scopes)
+
+        if credentials and credentials.valid:
+            # Make sure to exit early since we don't want to try to save these
+            # credentials to a cache file.
+            return credentials
+
     if auth_local_webserver is not None:
```

Comment:
[unrelated to this PR]
seems lines 336-366 are only relevant to the `if credentials is None` clause at line 366, and can be moved inside for better readability.